matrix_sdk_crypto/src/verification/mod.rs

```diff
@@ -12,13 +12,436 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+mod cache;
+mod event_enums;
 mod machine;
 mod requests;
 mod sas;
 
+use std::sync::Arc;
+
+use event_enums::OutgoingContent;
 pub use machine::VerificationMachine;
+use matrix_sdk_common::{
+    api::r0::keys::upload_signatures::Request as SignatureUploadRequest,
+    events::{
+        key::verification::{
+            cancel::{CancelCode, CancelEventContent, CancelToDeviceEventContent},
+            done::{DoneEventContent, DoneToDeviceEventContent},
+            Relation,
+        },
+        AnyMessageEventContent, AnyToDeviceEventContent,
+    },
+    identifiers::{DeviceId, EventId, RoomId, UserId},
+};
 pub use requests::VerificationRequest;
-pub use sas::{AcceptSettings, Sas, VerificationResult};
+pub use sas::{AcceptSettings, Sas};
+use tracing::{error, info, trace, warn};
+
+use crate::{
+    error::SignatureError,
+    olm::PrivateCrossSigningIdentity,
+    store::{Changes, CryptoStore, DeviceChanges},
+    CryptoStoreError, LocalTrust, ReadOnlyDevice, UserIdentities,
+};
+
+#[derive(Clone, Debug)]
+pub enum Verification {
```

Comment:
Non-exhaustive?

Reply:
Why?

Reply:
Because in the future more SAS versions will be added, and it's a breaking change to add an extra item to an enum.

Though I might be mistaken, ruma uses non-exhaustive types to not constantly have to make breaking changes when adding something to one of those enums, if rust-sdk will follow semantic versioning, I guess this'll be alright.

Reply:
Ruma probably uses non-exhaustiveness because the likelihood of new fields in the structs/enums is quite high, in contrast it's quite rare that we get a completely new verification flow so I'm fine with version bumps and breaking changes here.